src/components/Visualization.tsx

```diff
@@ -51,10 +53,55 @@ const Visualization = () => {
     shallow,
   );
 
-  const flows = useFlowsStore((state) => state.flows);
+  const { settings, setSettings } = useSettingsStore((state) => state);
+  const { setSourceCode } = useIntegrationSourceStore((state) => state);
```

Comment:
Now this could be like
```suggestion
  const { setSourceCode, setSyncedSourceCode } = useIntegrationSourceStore((state) => state);
```
@lordrip is there any benefit to do even like this?
```typescript
  const { setSourceCode, setSyncedSourceCode } = useIntegrationSourceStore(({setSourceCode, setSyncedSourceCode) => ({setSourceCode, setSyncedSourceCode}));
```

Reply:
thanks - updated

Reply:
> @lordrip is there any benefit to do even like this?

From the functional point of view, both should be the same, what would be different though is to use the `shallow` function when using the store, because otherwise this component will be rendered any time that a property changes, despite being relevant or not, so my recommendation would be, bring the least amount of data required, and do the composition in the function itself, so save some calculations.

```typescript
/**
* IMHO good as we only use what we need and the `shallow`
* function operates over 2 properties instead of the whole `state` object
*/
import { shallow } from 'zustand';

const { prop1, prop2 } = useMyStore((state) => ({
    prop1: state.prop1,
    prop2: state.prop1 / 1_000, // making calculations
}) , shallow);
```

```typescript
/**
* Same as before, but now we're adding overhead by checking whether
* the object or its properties changed
*/
import { shallow } from 'zustand';

const { prop1, prop2 } = useMyStore((state) => state, shallow);
```

```typescript
/**
* Definitively not good as we render anytime that something change
*/

const { prop1, prop2 } = useMyStore();
```